src/line_parser.py

```diff
@@ -1,149 +1,132 @@
 import argparse
-import json
 import cv2
 import random
 import numpy as np
 import re
+import check_letter
+from math import ceil
 
 
 class LineParser:
     def __init__(self, hashes):
         self.hashes = hashes
 
-    # Generates and image of a line of text
+    # Generates image of a line of text
     def parse_line(self, line):
         
         assert(len(line) > 0)
```

Comment:
include a try and catch exception. This will raise an assertion error if the len is less than 0. So at this point gui will fail with no prompts in frontend.

Reply:
No need for this, as parse_line_constrained never passes an empty line to parse_line (in the case of an empty document, an empty page outputted. Nevertheless, removed the assert statement from code to avoid confusion in the future.